### ISSUE TITLE
Reports generating twice

### ISSUE THREAD

**Issue by pisuke** [faucetsdn/daq#137]
Fix report output to generate only one report.
At the moment there is a bug that output two reports, 1 showing half of the information from the test and the other is a full report.

**Comment by grafnu** [comment]
Needs to be reproduced locally.  Francesco to provide more details on what the failure looks like.

**Comment by pisuke** [comment]
Steps to reproduce the problem.

git pull configurator branch of DAQ
https://github.com/grafnu/daq/tree/configurator

build DAQ

Configure in test_lab mode
https://github.com/grafnu/daq/blob/configurator/docs/test_lab.md

Configure Firebase UI

cmd/run -n

Click hold button in Firebase UI

check reports generated in inst/reports, there are one with data in it and one empty.

I've attached the two examples of reports generated as single run and saved into inst/report, one OK and one empty.


[drive-download-20190529T180412Z-001.zip](https://github.com/faucetsdn/daq/files/3233812/drive-download-20190529T180412Z-001.zip)

**Comment by sdrayton1066** [comment]
Reproduced issue on ubuntu-classic-2 : for a given timestamp, two reports are generated, in /daq/inst/reports, one containing data, one with a header but no data, as described above.

**Comment by grafnu** [comment]
I think the half-reports are either reports-in-progress (not complete yet), or from termnated run sessions.  At least all the ones I found were like that.  Either way, I updated the code in https://github.com/grafnu/daq/pull/150 to write out the report all at once at the end to avoid the partial version (it just won't be there at all). Either give that branch a try to see if it's fixed, or wait until that PR goes through.
